src/redux/reducers/language.js

```diff
@@ -36,22 +36,22 @@ const languageReducer = produce((state, action) => {
     case SET_CONSTANTS:
       parseLanguage(state.constants, action.value, CONSTANT);
       setConstants(state);
-      setPredicates(state);
-      setFunctions(state);
+      setPredicates(state, true);
+      setFunctions(state, true);
```

Comment:
Boolovskym argumentom sa vyhybajte, ako sa len da, su cestou do pekla, zneprehladnuju volajuci aj volany kod.

Ked sa vam zda, ze potrebujete boolovsky argument, v skutocnosti potrebujete dve funkcie. Neznamena to, ze budete kopirovat kod, ale ze budete z jednej volat druhu alebo budete z oboch volat nejake pomocne funkcie.

Ujasnite si, co sa tu ma diat:
- niekedy treba parsovat a validovat (obsluha akcii z grafu),
- inokedy treba iba validovat (obsluha akcii pri zmene textovych inputov mnozin symbolov konstant, predikatov, funkcii).

Validacne funkcie z `functions/validate` maju priserny interface, ktory vam zabranuje volat ich jednoducho. A `setConstats` a spol. navyse robia nedobre testy na `.errorMessage`, ktore sposobuju, ze validacne chyby sa daju zrusit iba pridavanim medzier do vstupov.

Takze odporucam:
- Upravte `validateConstants` a spol., aby proste zobrali `state` a rovno nastavili prislusny `.errorMessage`. Stale tam bude vela viac-menej kopirovaneho kodu, co mi vadi, ale nie je to take podstatne.
- Vsade, kde netreba reparsovat, volajte rovno takto upravenu validaciu.
- Vsade, kde treba aj parsovat, volajte `setConstants` a spol., ktore sparsuju a ak parsovanie zbehne, tak zavolaju validaciu. Idealne `setConstants` premenujte na `parseAndValidateConstants`. Pri obsluhe `SET_CONSTANTS` potom staci priradit `action.value` do `state.constants.value` a zavolat `parseAndValidateConstants` (a potom ostatne `parseAndValidate…`).